services/business/checkers/route_rules/version_presence_checker.go

```diff
@@ -0,0 +1,82 @@
+package route_rules
+
+import (
+	"reflect"
+	"strconv"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/services/models"
+)
+
+type VersionPresenceChecker struct {
+	Namespace string
+	PodList   []v1.Pod
+	RouteRule kubernetes.IstioObject
+}
+
+func (checker VersionPresenceChecker) Check() ([]*models.IstioCheck, bool) {
+	valid := true
+	validations := make([]*models.IstioCheck, 0)
+
+	slice := reflect.ValueOf(checker.RouteRule.GetSpec()["route"])
+	if slice.Kind() != reflect.Slice {
+		return validations, valid
+	}
+
+	for i := 0; i < slice.Len(); i++ {
+		route := slice.Index(i).Interface().(map[string]interface{})
```

Comment:
I think this cast should be protected, as a typo in the routerule yaml is possible and you might not cast this, so you would get then a nice panic.
A simple route, ok := ... should help.